core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -31,24 +31,16 @@ import org.apache.spark.sql.types.{DataType, Decimal, DecimalType, StructType}
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.types.UTF8String
 
-trait CompactExecRowToMutableRow extends ResultWasNull {
+abstract class CompactExecRowToMutableRow extends ResultNullHolder {
```

Comment:
Also CompactExecRowToMutableRow is not used by row table scan at all. It is only used when inserting as of now. The generated code directly gets handle to AbstractCompactExecRow (either via direct region iterator or peeking inside EmbedResultSet when there are index filters but no projections), then invokes its required getAs* methods passing a common ResultNullHolder object to check whether the result was null or not (the setWasNull is ultimately invoked by RowFormatter is required).